#!/usr/bin/env python
PACKAGE = "kinect_slam"
 
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("harris_window_size", int_t, 0, "harris_window_size", 6, 1, 15)
gen.add("harris_response_threshold", double_t, 0, "harris_response_threshold", 0.01, 0.0, 0.1)
gen.add("harris_anms_flag", bool_t, 0, "harris_anms_flag", True);
gen.add("harris_anms_radius", int_t, 0, "harris_anms_radius", 2, 1, 20)
gen.add("harris_fix_number_flag", bool_t, 0, "harris_fix_number_flag", True);
gen.add("harris_number_of_points", int_t, 0, "harris_number_of_points", 1000, 1, 5000)
gen.add("harris_blur_size", int_t, 0, "harris_blur_size", 4, 1, 20)
gen.add("harris_blur_variance", int_t, 0, "harris_blur_variance", 9, 1, 20)

gen.add("descriptor_smooth_size", int_t, 0, "descriptor_smooth_size", 6, 1, 10)
gen.add("descriptor_brief_size", int_t, 0, "descriptor_brief_size", 1, 1, 3)
gen.add("descriptor_patch_size", int_t, 0, "descriptor_patch_size", 19, 15, 30)

gen.add("vertical_offset", int_t, 0, "vertical_offset", 30, 0, 300)
gen.add("min_horizontal_threshold", int_t, 0, "min_horizontal_threshold", 3, 0, 9)
gen.add("max_horizontal_threshold", int_t, 0, "max_horizontal_threshold", 80, 10, 300)

gen.add("matcher_dist_threshold", double_t, 0, "matcher_dist_threshold", 0.85, 0, 1)

gen.add("ransac_n", int_t, 0, "ransac_n", 15, 1, 100)
gen.add("tf_threshold", double_t, 0, "tf_threshold", 500, 0, 5000)

exit(gen.generate(PACKAGE, "kinect_slam", "KinectSLAM"))
